e2e/cypress/e2e/sidebar/quickstart-sidebar.cy.ts

```diff
@@ -0,0 +1,415 @@
+/**
+ * This test suite validates that sidebar filtering and tagger menu functionality works in the fiftyone app with quickstart dataset.
+ */
+
+const selectSamplesInGrid = (indices: number[]) => {
+  indices.forEach((i) => {
+    cy.get("[data-cy=looker]")
+      .eq(i)
+      .within(() => {
+        cy.get("input").click({ force: true });
+      });
+  });
+};
+
+describe("Sidebar filter", () => {
+  context("quickstart dataset filter", () => {
+    before(() => {
+      cy.executePythonCode(
+        `
+            import fiftyone as fo
+            import fiftyone.zoo as foz
+            quickstart_dataset = foz.load_zoo_dataset("quickstart", max_samples=50, dataset_name="quickstart-test")
+            quickstart_dataset.persistent = True
+          `
+      );
+    });
+
+    beforeEach(() => {
+      cy.waitForGridToBeVisible("quickstart-test");
+    });
+
+    it("should show all default sidebar groups by names", () => {
+      const defaultGroups = ["TAGS", "METADATA", "LABELS", "PRIMITIVES"];
+      defaultGroups.forEach((g) => {
+        cy.get(`[data-cy=sidebar-group-${g}]`).should("exist");
+      });
+    });
+
+    it("should be able to filter by ground_truth label", () => {
+      // ground_truth selector should be visible and expandable
+      cy.get("[data-cy=sidebar-field-ground_truth]").should("be.visible");
+      cy.get("[data-cy=sidebar-field-arrow-ground_truth]").should("be.visible");
+      cy.get("[data-cy=sidebar-field-arrow-ground_truth]").click();
+
+      // entry count should be 199 before filtering
+      cy.contains("199", { timeout: 5000 }).should("be.visible");
+      cy.get(
+        "[data-cy=categorical-filter-ground_truth\\.detections\\.label]"
+      ).should("exist");
+      cy.get("[data-cy=sidebar-field-container-ground_truth").within(() => {
+        cy.get("[data-cy=entry-count-all]").should("have.text", 199);
+      });
+
+      // default filter mode of select "cat" label, expect 7 samples out 50 samples
+      cy.get('[data-cy="selector-\\+ filter by label"]').should("exist");
+      cy.get("[data-cy=selector-div-ground_truth\\.detections\\.label]").within(
+        () => {
+          // do not use cy.get("input").type("cat{enter}");
+          // that causes the test to be very flasky
+          cy.get("input").type("cat");
+          cy.get("input").should("have.value", "cat");
+          cy.wait(300);
```

Comment:
Is it possible to avoid time based wait?

Reply:
Or if you _have_ to use it, add a comment on why.